fix: enter unreachable when discarding HBH Option with multicast destination address

thvdveld opened the pull request:
It was previously possible to enter an unreachable path when the HBH option type was 0b11 and the destination address was multicast. This is now fixed in the interface.

I also changed the way an HBH option type is parsed. Only the 2 highest order bits indicate what action must be taken, meaning that we can't have an unknown state.

thvdveld committed:
fix: enter unreachable HBH Discard multicast addr

It was previously possible to enter an unreachable path when the HBH
option type was 0b11 and the destination address was multicast. This is
now fixed in the interface.

I also changed the way an HBH option type is parsed. The option type is
only 2 bits, meaning that we can't have an unknown state.

Dirbaio commented on src/iface/interface/ipv6.rs:
```diff
@@ -288,12 +288,11 @@ impl InterfaceInner {
+                        Ipv6OptionFailureType::DiscardSendUnicast => {
+                            if !ipv6_repr.dst_addr.is_multicast() {
+                                return HopByHopResponse::Discard(param_problem());
+                            }
```
we still have to return here if the sender is multicast, to discard the packet, right?

thvdveld replied:
Yes, indeed, I misread this in the RFC. I thought only the option needs to be discarded, but like you say, the full packet needs to be discarded, without sending an ICMP parameter problem back.

thvdveld committed:
fix: IPv6-hbb: discard as indicated by hbh type

Discard the packet if the Hop-by-Hop header type is 0b11 and the packet
destination address is multicast.
